core/src/main/java/com/venia/core/models/commerce/servlets/CatalogPageExceptionFilter.java

```diff
@@ -0,0 +1,144 @@
+package com.venia.core.models.commerce.servlets;
+
+import com.adobe.cq.commerce.core.components.models.common.SiteStructure;
+import com.adobe.cq.commerce.core.components.models.product.Product;
+import com.adobe.cq.commerce.core.components.models.productlist.ProductList;
+import com.adobe.cq.commerce.core.components.models.retriever.AbstractCategoryRetriever;
+import com.adobe.cq.commerce.core.components.models.retriever.AbstractProductRetriever;
+import com.day.cq.wcm.api.Page;
+import com.day.cq.wcm.api.PageManager;
+import com.day.cq.wcm.api.PageManagerFactory;
+import com.venia.core.models.commerce.services.CommerceComponentModelFinder;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.SlingHttpServletResponse;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.scripting.SlingBindings;
+import org.apache.sling.scripting.core.ScriptHelper;
+import org.osgi.framework.BundleContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.servlet.*;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+@Component(
+        service = {Filter.class},
+        property = {
+                "sling.filter.scope=REQUEST",
+                "sling.filter.resourceTypes=cq:Page",
+                "sling.filter.resourceTypes=core/cif/components/structure/page/v1/page",
+                "sling.filter.resourceTypes=core/cif/components/structure/page/v2/page",
+                "sling.filter.resourceTypes=core/cif/components/structure/page/v3/page",
+                "sling.filter.extensions=html",
+                "sling.filter.extensions=json",
+                "sling.filter.resource.pattern=/content(/.+)?",
+                "service.ranking:Integer=-4000"
+        })
+public class CatalogPageExceptionFilter implements Filter {

```

Comment:
A better name would be here CatalogPageErrorFilter because here we are dealing with errors objects of the GraphQL response and not Java exceptions.

Reply:
Done